From 5792e237d31a54e0979a0c663c06c01b46e3f152 Mon Sep 17 00:00:00 2001
From: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Date: Thu, 21 Oct 2021 11:50:40 +0300
Subject: [PATCH 08/11] arch: s32: fix cmu fc/fm units

This patch corrects the definitions for the clock monitor unit
frequency monitor/frequency check CMU FM/FC. These modules can be
used to check that a clock is in the specified range(FC) or to read
the clock's value(FM).

Issue: ALB-7938

Upstream-Status: Pending 

Signed-off-by: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/cpu/armv8/s32/s32-gen1/include/cmu.h |  2 ++
 arch/arm/cpu/armv8/s32/s32-gen1/s32g2_cmu.c   | 12 +++++++++++-
 arch/arm/cpu/armv8/s32/s32-gen1/s32g3_cmu.c   | 14 ++++++++++++--
 arch/arm/cpu/armv8/s32/s32-gen1/s32r45_cmu.c  | 10 ++++++++++
 4 files changed, 35 insertions(+), 3 deletions(-)

diff --git a/arch/arm/cpu/armv8/s32/s32-gen1/include/cmu.h b/arch/arm/cpu/armv8/s32/s32-gen1/include/cmu.h
index 3b1c18f1fe..7ace47dea7 100644
--- a/arch/arm/cpu/armv8/s32/s32-gen1/include/cmu.h
+++ b/arch/arm/cpu/armv8/s32/s32-gen1/include/cmu.h
@@ -60,6 +60,7 @@ enum cmu_fc_clk {
 	XBAR_CLK_M7_0,
 	XBAR_CLK_M7_1,
 	XBAR_CLK_M7_2,
+	XBAR_CLK_M7_3,
 	XBAR_DIV3_CLK,
 	SERDES_REF_CLK,
 	PER_CLK,
@@ -89,6 +90,7 @@ enum cmu_fc_clk {
 	PFE_MAC_2_RX_CLK,
 	FTM_0_REF_CLK,
 	FTM_1_REF_CLK,
+	FLEXRAY_PE_CLK,
 };
 
 struct cmu {
diff --git a/arch/arm/cpu/armv8/s32/s32-gen1/s32g2_cmu.c b/arch/arm/cpu/armv8/s32/s32-gen1/s32g2_cmu.c
index d76979b844..7302d615b7 100644
--- a/arch/arm/cpu/armv8/s32/s32-gen1/s32g2_cmu.c
+++ b/arch/arm/cpu/armv8/s32/s32-gen1/s32g2_cmu.c
@@ -8,29 +8,39 @@
 
 static struct cmu cmu_blocks[] = {
 	FIRC_PERIPH_CMU_FC(0, FXOSC_CLK, FXOSC_FREQ),
+
 	FXOSC_PERIPH_CMU_FM(1, FIRC_CLK, FIRC_FREQ),
 	FXOSC_PERIPH_CMU_FM(2, SIRC_CLK, SIRC_FREQ),
 	FXOSC_PERIPH_CMU_FM(3, FTM_0_REF_CLK, 40),
 	FXOSC_PERIPH_CMU_FM(4, FTM_1_REF_CLK, 40),
-	FIRC_PERIPH_CMU_FC(0, FXOSC_CLK, FXOSC_FREQ),
+
 	FIRC_PERIPH_CMU_FC(5, XBAR_DIV3_CLK, 133.33),
 	FIRC_PERIPH_CMU_FC(6, XBAR_CLK_M7_0, 400),
+
 	FXOSC_PERIPH_CMU_FC(7, XBAR_DIV3_CLK, 133.33),
+
 	FIRC_PERIPH_CMU_FC(8, XBAR_CLK_M7_1, 400),
 	FIRC_PERIPH_CMU_FC(9, XBAR_CLK_M7_2, 400),
 	FIRC_PERIPH_CMU_FC(10, PER_CLK, 80),
+
 	FXOSC_PERIPH_CMU_FC(11, SERDES_REF_CLK, 125),
+	FXOSC_PERIPH_CMU_FC(12, FLEXRAY_PE_CLK, 40),
 	FXOSC_PERIPH_CMU_FC(13, CAN_PE_CLK, 80),
 	FXOSC_PERIPH_CMU_FC(14, GMAC_0_TX_CLK, 125),
 	FXOSC_PERIPH_CMU_FC(15, GMAC_TS_CLK, 200),
 	FXOSC_PERIPH_CMU_FC(16, LIN_CLK, 125),
 	FXOSC_PERIPH_CMU_FC(17, QSPI_1X_CLK, 200),
 	FXOSC_PERIPH_CMU_FC(18, SDHC_CLK, 400),
+
 	FIRC_PERIPH_CMU_FC(20, DDR_CLK, 666.66),
+
 	FXOSC_PERIPH_CMU_FC(21, GMAC_0_RX_CLK, 125),
 	FXOSC_PERIPH_CMU_FC(22, SPI_CLK, 100),
+
 	FXOSC_PERIPH_CMU_FC(27, A53_CORE_CLK, 1000),
+
 	FIRC_PERIPH_CMU_FC(28, A53_CORE_CLK, 1000),
+
 	FXOSC_PERIPH_CMU_FC(39, PFE_SYS_CLK, 300),
 	FXOSC_PERIPH_CMU_FC(46, PFE_MAC_0_TX_CLK, 312.5),
 	FXOSC_PERIPH_CMU_FC(47, PFE_MAC_0_RX_CLK, 312.5),
diff --git a/arch/arm/cpu/armv8/s32/s32-gen1/s32g3_cmu.c b/arch/arm/cpu/armv8/s32/s32-gen1/s32g3_cmu.c
index cc7fbc5d02..d71d9eb00c 100644
--- a/arch/arm/cpu/armv8/s32/s32-gen1/s32g3_cmu.c
+++ b/arch/arm/cpu/armv8/s32/s32-gen1/s32g3_cmu.c
@@ -6,29 +6,39 @@
 
 static struct cmu cmu_blocks[] = {
 	FIRC_PERIPH_CMU_FC(0, FXOSC_CLK, FXOSC_FREQ),
+
 	FXOSC_PERIPH_CMU_FM(1, FIRC_CLK, FIRC_FREQ),
 	FXOSC_PERIPH_CMU_FM(2, SIRC_CLK, SIRC_FREQ),
 	FXOSC_PERIPH_CMU_FM(3, FTM_0_REF_CLK, 40),
 	FXOSC_PERIPH_CMU_FM(4, FTM_1_REF_CLK, 40),
-	FIRC_PERIPH_CMU_FC(0, FXOSC_CLK, FXOSC_FREQ),
+
+	FIRC_PERIPH_CMU_FC(5, XBAR_DIV3_CLK, 133),
 	FIRC_PERIPH_CMU_FC(6, XBAR_CLK_M7_0, 399.96),
+
 	FXOSC_PERIPH_CMU_FC(7, XBAR_DIV3_CLK, 133.32),
+
 	FIRC_PERIPH_CMU_FC(8, XBAR_CLK_M7_1, 399.96),
 	FIRC_PERIPH_CMU_FC(9, XBAR_CLK_M7_2, 399.96),
+	FIRC_PERIPH_CMU_FC(24, XBAR_CLK_M7_3, 399.96),
 	FIRC_PERIPH_CMU_FC(10, PER_CLK, 80),
-	FIRC_PERIPH_CMU_FC(5, XBAR_DIV3_CLK, 133.32),
+
 	FXOSC_PERIPH_CMU_FC(11, SERDES_REF_CLK, 125),
+	FXOSC_PERIPH_CMU_FC(12, FLEXRAY_PE_CLK, 40),
 	FXOSC_PERIPH_CMU_FC(13, CAN_PE_CLK, 80),
 	FXOSC_PERIPH_CMU_FC(14, GMAC_0_TX_CLK, 125),
 	FXOSC_PERIPH_CMU_FC(15, GMAC_TS_CLK, 200),
 	FXOSC_PERIPH_CMU_FC(16, LIN_CLK, 125),
 	FXOSC_PERIPH_CMU_FC(17, QSPI_1X_CLK, 200),
 	FXOSC_PERIPH_CMU_FC(18, SDHC_CLK, 400),
+
 	FIRC_PERIPH_CMU_FC(20, DDR_CLK, 800),
+
 	FXOSC_PERIPH_CMU_FC(21, GMAC_0_RX_CLK, 125),
 	FXOSC_PERIPH_CMU_FC(22, SPI_CLK, 100),
 	FXOSC_PERIPH_CMU_FC(27, A53_CORE_CLK, 1300),
+
 	FIRC_PERIPH_CMU_FC(28, A53_CORE_CLK, 1300),
+
 	FXOSC_PERIPH_CMU_FC(39, PFE_SYS_CLK, 300),
 	FXOSC_PERIPH_CMU_FC(46, PFE_MAC_0_TX_CLK, 312.5),
 	FXOSC_PERIPH_CMU_FC(47, PFE_MAC_0_RX_CLK, 312.5),
diff --git a/arch/arm/cpu/armv8/s32/s32-gen1/s32r45_cmu.c b/arch/arm/cpu/armv8/s32/s32-gen1/s32r45_cmu.c
index 9f282c9b2b..c47685a9e4 100644
--- a/arch/arm/cpu/armv8/s32/s32-gen1/s32r45_cmu.c
+++ b/arch/arm/cpu/armv8/s32/s32-gen1/s32r45_cmu.c
@@ -7,28 +7,38 @@
 
 static struct cmu cmu_blocks[] = {
 	FIRC_PERIPH_CMU_FC(0, FXOSC_CLK, FXOSC_FREQ),
+
 	FXOSC_PERIPH_CMU_FM(1, FIRC_CLK, FIRC_FREQ),
 	FXOSC_PERIPH_CMU_FM(2, SIRC_CLK, SIRC_FREQ),
 	FXOSC_PERIPH_CMU_FM(3, FTM_0_REF_CLK, 40),
 	FXOSC_PERIPH_CMU_FM(4, FTM_1_REF_CLK, 40),
+
 	FIRC_PERIPH_CMU_FC(5, XBAR_DIV3_CLK, 133.33),
 	FIRC_PERIPH_CMU_FC(6, XBAR_CLK_M7_0, 400),
+
 	FXOSC_PERIPH_CMU_FC(7, XBAR_DIV3_CLK, 133.33),
+
 	FIRC_PERIPH_CMU_FC(8, XBAR_CLK_M7_1, 400),
 	FIRC_PERIPH_CMU_FC(9, XBAR_CLK_M7_2, 400),
 	FIRC_PERIPH_CMU_FC(10, PER_CLK, 80),
+
 	FXOSC_PERIPH_CMU_FC(11, SERDES_REF_CLK, 125),
+	FXOSC_PERIPH_CMU_FC(12, FLEXRAY_PE_CLK, 40),
 	FXOSC_PERIPH_CMU_FC(13, CAN_PE_CLK, 80),
 	FXOSC_PERIPH_CMU_FC(14, GMAC_0_TX_CLK, 125),
 	FXOSC_PERIPH_CMU_FC(15, GMAC_TS_CLK, 200),
 	FXOSC_PERIPH_CMU_FC(16, LIN_CLK, 125),
 	FXOSC_PERIPH_CMU_FC(17, QSPI_1X_CLK, 133.33),
 	FXOSC_PERIPH_CMU_FC(18, SDHC_CLK, 400),
+
 	FIRC_PERIPH_CMU_FC(20, DDR_CLK, 800),
+
 	FXOSC_PERIPH_CMU_FC(21, GMAC_0_RX_CLK, 125),
 	FXOSC_PERIPH_CMU_FC(22, SPI_CLK, 100),
 	FXOSC_PERIPH_CMU_FC(27, A53_CORE_CLK, 800),
+
 	FIRC_PERIPH_CMU_FC(28, A53_CORE_CLK, 800),
+
 	FXOSC_PERIPH_CMU_FC(38, ACCEL3_CLK, 600),
 	FXOSC_PERIPH_CMU_FC(39, ACCEL4_CLK_0, 400),
 	FXOSC_PERIPH_CMU_FC(40, ACCEL4_CLK_0, 400),
-- 
2.17.1

